app/assets/javascripts/discourse/app/components/async-content.gjs

```diff
@@ -1,13 +1,23 @@
 import Component from "@glimmer/component";
 import { cached } from "@glimmer/tracking";
+import { hash } from "@ember/helper";
+import { htmlSafe } from "@ember/template";
 import { TrackedAsyncData } from "ember-async-data";
 import { Promise as RsvpPromise } from "rsvp";
+import { eq } from "truth-helpers";
 import ConditionalLoadingSpinner from "discourse/components/conditional-loading-spinner";
+import FlashMessage from "discourse/components/flash-message";
 import { popupAjaxError } from "discourse/lib/ajax-error";
 import discourseDebounce from "discourse/lib/debounce";
+import { bind } from "discourse/lib/decorators";
 import { INPUT_DELAY } from "discourse/lib/environment";
+import { extractErrorInfo } from "../lib/ajax-error";
+
+const ERROR_MODES = ["flash", "popup"];
+const DEFAULT_ERROR_MODE = "flash";
 
 export default class AsyncContent extends Component {
+  errorMode = this.args.errorMode ?? DEFAULT_ERROR_MODE;
```

Comment:
Would be better to make this a getter, in case the `errorMode` argument changes after the component is initialized